### PR TITLE
Get homo lumo energies

### PR DESCRIPTION
**Pull request by twsleight**
Adding another function and a test to get the HOMO and LUMO energies out of a .out script.

also:

.travis.yml and setup.py files included to make sure that it worked on Travis.

requirements.txt simplified

### CODE REVIEW

File: tests/test_gethlenergy.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+filename = 'coumarin_gas.out'
+
+def test_gethlenergy():
+    path = os.path.abspath(__file__)
+    dir_path = os.path.dirname(path)
+
+    [h, l] = gethlenergy(dir_path, filename)
+    print(h,l)
+    assert(h == -0.29586 and l == -0.04159)
```
Comment (paulboone) [inline tests/test_gethlenergy.py]: hl = gethlenergy(dir_path, filename)
assert hl == [-0.29586, -0.04159]

Comment (paulboone) [reply tests/test_gethlenergy.py]: shouldn't need print

File: ames_qsar/gethlenergy.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    termFlag = 0
+    errorFlag = 0
+    a = 0
+    b = 0
+    #initialize a holding variable
+    prevLine = []
+    os.chdir(dir_path)
+
+    with open(filename)as searchfile:
+
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py:16]: ) as

File: ames_qsar/gethlenergy.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+def gethlenergy(dir_path, filename):
+
+    #initilalize the error codes as ok
+    termFlag = 0
+    errorFlag = 0
+    a = 0
+    b = 0
+    #initialize a holding variable
+    prevLine = []
+    os.chdir(dir_path)
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: don't chdir

File: ames_qsar/gethlenergy.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#gethlenergy will take a filepath and a file name
+#and return the HOMO and LUMO energies.
+
+#this will also need a test
+import os
+
+def gethlenergy(dir_path, filename):
+
+    #initilalize the error codes as ok
+    termFlag = 0
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: term_flag

File: ames_qsar/gethlenergy.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            if sep and 'Alpha  occ. eigenvalues --' in prevLine:
+
+                b1 = (right[:56])
+                b1 = b1. split()
+                b = float(b1[0])
+
+                left,sep,right=prevLine.partition('Alpha  occ. eigenvalues --')
+
+                a1 = (right[:56])
+                a1 = a1. split()
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: '. '

File: ames_qsar/gethlenergy.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            #***************************************************************
+
+            left,sep,right=line.partition('Alpha virt. eigenvalues --')
+
+            if sep and 'Alpha  occ. eigenvalues --' in prevLine:
+
+                b1 = (right[:56])
+                b1 = b1. split()
+                b = float(b1[0])
+
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py:33]: b = float(right[:56].split()[0])

File: ames_qsar/gethlenergy.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        for line in searchfile:
+            #error checking
+            #the normal termination flag should turn to 1
+            if 'Normal termination of Gaussian 09' in line:
+                termFlag = 1
+            if 'slurmstepd: error:' in line:
+                errorFlag = 1
+            #***************************************************************
+
+            left,sep,right=line.partition('Alpha virt. eigenvalues --')
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: whitespace 'left,sep,right'

File: ames_qsar/gethlenergy.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                b1 = b1. split()
+                b = float(b1[0])
+
+                left,sep,right=prevLine.partition('Alpha  occ. eigenvalues --')
+
+                a1 = (right[:56])
+                a1 = a1. split()
+                a = float(a1[-1])
+
+            prevLine2 = prevLine
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: delete?

File: ames_qsar/gethlenergy.py (hunk 7 above)
Comment (paulboone) [reply ames_qsar/gethlenergy.py]: if 'Alpha virt. eigenvalues --' in line and 'Alpha  occ. eigenvalues --' in prev_line:

File: ames_qsar/gethlenergy.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            prevLine = line
+        #*************************
+        #end of the searchfile loop
+
+        if termFlag == 1 and errorFlag == 0:
+        #********************
+            print(a, b)
+        #********************
+        else:
+            print(filename, ' did not terminate correctly')
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: raise an exception?

File: ames_qsar/gethlenergy.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+            prevLine2 = prevLine
+            prevLine = line
+        #*************************
+        #end of the searchfile loop
+
+        if termFlag == 1 and errorFlag == 0:
+        #********************
+            print(a, b)
+        #********************
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: consider removing the print?

File: ames_qsar/gethlenergy.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+        #These fences indicate the where the loop ends
+        #************************
+        for line in searchfile:
+            #error checking
+            #the normal termination flag should turn to 1
+            if 'Normal termination of Gaussian 09' in line:
+                termFlag = 1
+            if 'slurmstepd: error:' in line:
+                errorFlag = 1
```
Comment (monshap) [inline ames_qsar/gethlenergy.py]: Raise error here, so you don't run through the rest of the code.

File: ames_qsar/gethlenergy.py [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    a = 0
+    b = 0
+    #initialize a holding variable
+    prevLine = []
+    os.chdir(dir_path)
+
+    with open(filename)as searchfile:
+
+        #These fences indicate the where the loop ends
+        #************************
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: #******* comment => generally don't do this

File: setup.py [hunk 13]
```diff
@@ -0,0 +1,14 @@
+from setuptools import setup
+
+with open('requirements.txt', 'r') as f:
+    requirements = f.read().splitlines()
+
```
Comment (paulboone) [inline setup.py]: replace this with just putting packages directly into setup.py

**Review by paulboone — CHANGES_REQUESTED** [review]
Changes to the new homolumo functions!

**Comment by twsleight** [comment]
updates made and force pushed

**Review by bayerl — APPROVED** [review]

File: ames_qsar/gethlenergy.py [hunk 14]
```diff
@@ -13,9 +13,8 @@ def gethlenergy(dir_path, filename):
     b = 0
     #initialize a holding variable
     prevLine = []
-    os.chdir(dir_path)
 
-    with open(filename) as searchfile:
+    with open((dir_path + '/' + filename)) as searchfile:
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: os.path.join(dir_path, filename)

Comment (paulboone) [reply ames_qsar/gethlenergy.py]: don't do this here, but you should know about this: "%s/%s" % (dir_path, filename)

File: ames_qsar/gethlenergy.py [hunk 15]
```diff
@@ -41,5 +40,7 @@ def gethlenergy(dir_path, filename):
 
             prevLine = line
         #end of the searchfile loop
+        if term_flag ==0:
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: term_flag => normal_termination or terminated_normally:
terminated_normally = True
if not terminated_normally:

File: ames_qsar/gethlenergy.py [hunk 16]
```diff
@@ -28,15 +28,15 @@ def gethlenergy(dir_path, filename):
 
             if sep and 'Alpha  occ. eigenvalues --' in prevLine:
 
-                b1 = (right[:56])
-                b1 = b1.split()
-                b = float(b1[0])
+                b = float((right[:56]).split()[0])
+                #b1 = b1.split()
+                #b = float(b1[0])
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: delete, not comment

File: ames_qsar/gethlenergy.py [hunk 17]
```diff
@@ -26,17 +26,15 @@ def gethlenergy(dir_path, filename):
 
             left,sep,right=line.partition('Alpha virt. eigenvalues --')
 
+            #The Alpha  occ. and Alpha virt lines should
+            #be found right next to each other
             if sep and 'Alpha  occ. eigenvalues --' in prevLine:
 
                 b = float((right[:56]).split()[0])
-                #b1 = b1.split()
-                #b = float(b1[0])
```
Comment (paulboone) [inline ames_qsar/gethlenergy.py]: this could be squashed

File: setup.py [hunk 18]
```diff
@@ -1,8 +1,5 @@
 from setuptools import setup
 
-with open('requirements.txt', 'r') as f:
-    requirements = f.read().splitlines()
-
```
Comment (paulboone) [inline setup.py:5]: squash with prior requirement changes

**Review by paulboone — CHANGES_REQUESTED** [review]

**Comment by twsleight** [comment]
updates from Thursday rebased and force pushed

**Comment by twsleight** [comment]
merged the getHOMOLUMOenergies branch with master so the commits updated. Resolved the merge conflicts locally and force pushed, which updated the PR. This seems to have eliminated the merge conflicts.

**Pull request merged**